tests/main/regression-home-snap-root-owned/task.yaml

```diff
@@ -0,0 +1,19 @@
+summary: Regression test that ensures that $HOME/snap is not root owned for sudo commands
+
+systems: [-ubuntu-core-16-64]
+
+prepare: |
+    # ensure we have no snap user data directory yet
+    rm -rf /home/test/snap
+    snap install test-snapd-tools
+
+execute: |
+    # run a snap command via sudo
+    su -l -c "sudo /snap/bin/test-snapd-tools.env" test
+
+    # and verify that the dirs have the right owner
```

Comment:
Can we please also check that `/root/snap` exists and is owned by root?

Otherwise this doesn't really check that we used the home directory of the root user IMHO